src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs

```diff
@@ -78,14 +78,15 @@ private async Task<HttpResponseMessage> SendRetryAsync(HttpResponseMessage respo
             int retryCount = 0;
             TimeSpan cumulativeDelay = TimeSpan.Zero;
 
-            while (retryCount < RetryOption.MaxRetry)
+            // Drain response content to free connections. Need to perform this
+            // before retry attempt and before the TooManyRetries ServiceException.
+            if (response.Content != null)
```

Comment:
now, we are not draining in the loop. I think this needs to happen after the loop, but before the ServiceException.